esp/public/media/scripts/ajaxschedulingmodule/ESP/Cell.js

```diff
@@ -96,13 +100,175 @@ function Cell(el, section, room_id, timeslot_id, matrix) {
+                return this.cellColors.HSLToRGB(0,100,100 - 10 * n_req); // Color red based on number of resource requests
+            case "hungry":
+                // If this section overlaps with lunch, count how many teachers are teaching other sections that overlap with the rest of the lunch time timeslots
+                var today = undefined;
+                var n_teachers = 0;
+                // Find which day, if any, of lunch constraints to worry about (we only care if this section overlaps with lunch at all)
+                day_loop:
+                    for(var day in this.matrix.timeslots.lunch_timeslots){
+                        for(var ts of this.matrix.timeslots.lunch_timeslots[day].map(x => x.id)){
+                            if(scheduleAssignment.timeslots.includes(ts)){
+                                today = day;
+                                break day_loop;
+                            }
+                        }
+                    }
+                if(today){
+                    // Figure out which teachers, if any, have sections that overlap with all of the lunch constraints for that day
+                    teacher_loop:
+                        for(var teacher of section.teacher_data){
+                            var lunch_ids = this.matrix.timeslots.lunch_timeslots[today].map(x => x.id);
+                            // Address this section separately, since it might be a ghost section (and therefore not in sections.scheduleAssignments)
+                            for(var ts of scheduleAssignment.timeslots){
+                                if(lunch_ids.includes(ts)){
+                                    lunch_ids.splice(lunch_ids.indexOf(ts), 1)
+                                }
+                                if(lunch_ids.length == 0){
+                                    n_teachers += 1;
+                                    continue teacher_loop;
+                                }
+                            }
+                            for(var sec of teacher.sections){
+                                for(var ts of this.matrix.sections.scheduleAssignments[sec].timeslots){
+                                    if(lunch_ids.includes(ts)){
+                                        lunch_ids.splice(lunch_ids.indexOf(ts), 1)
+                                    }
+                                    if(lunch_ids.length == 0){
+                                        n_teachers += 1;
+                                        continue teacher_loop;
+                                    }
+                                }
+                            }
+                        }
+                }
+                if(n_teachers == 0) return grey;
+                return this.cellColors.HSLToRGB(0,100,100 - 10 * n_teachers); // Color red based on number of teachers
+            case "double_booked":
+                // Count how many teachers are teaching other sections at the same time as this one
```

Comment:
is this referring to coteachers of a class? I think that could be more clear if so.